evm/v0.5/src/helpers.ts

```diff
@@ -650,10 +901,21 @@ export function findEventIn(
 }
 
 /**
- * Calculate six months from the current date in seconds
+ * Compute function signature hash for a function on an ethersjs Contract instance
+ * @param contract ethersjs contract
+ * @param functionName The function name
  */
-export function sixMonthsFromNow(): ethers.utils.BigNumber {
-  return ethers.utils.bigNumberify(
-    Math.round(Date.now() / 1000.0) + 6 * 30 * 24 * 60 * 60,
-  )
+export function getFunctionSignature(
```

Comment:
I think this function is unnecessary indirection, we can directly get the sighash. You can make the access a little shorter by going through the contract interface, which avoids having to do `.interface`. Another reason for not needing this is that we lose type safety by not checking what function names are valid or not, which is what `myContract.functions.myFunc` would do for you.